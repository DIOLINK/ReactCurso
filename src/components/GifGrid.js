import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p className="animate__pulse">Loading...</p>}
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem {...img} key={img.id} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
