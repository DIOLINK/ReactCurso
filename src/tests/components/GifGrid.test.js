import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
describe('Probando GifGrid.js', () => {
  const category = 'One Punch';
  useFetchGifs.mockReturnValue({
    data: [],
    loading: true,
  });
  const wrapper = shallow(<GifGrid category={category} />);
  it('Debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Debe de tener un h3 con la categoria', () => {
    expect(wrapper.find('h3').text().trim()).toBe(category);
  });
  it('Debe de tener un parrafo con la categoria', () => {
    expect(wrapper.find('p').text().trim()).toBe('Loading...');
  });
  it('Debe tener la data', () => {
    const gifs = [
      { id: 'ABC', url: 'https://google.com', title: 'Cualquier cosa' },
      { id: '123', url: 'https://google.com', title: 'Cualquier cosa Ro' },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
