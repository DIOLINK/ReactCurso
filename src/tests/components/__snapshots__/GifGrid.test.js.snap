// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifGrid.js Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p
    className="animate__pulse"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
